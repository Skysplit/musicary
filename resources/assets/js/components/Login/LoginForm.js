import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { Button } from 'react-md';

import TextField from '../Forms/Inputs/TextField';

export const LoginForm = ({
  handleSubmit,
}) => (
  <form onSubmit={handleSubmit}>
    <div>
      <Field name="email" label="email" component={TextField} type="text" />
    </div>
    <div>
      <Field name="password" label="password" component={TextField} type="password" />
    </div>
    <div>
      <Button type="submit" label="Sign in" primary raised />
    </div>
  </form>
);

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({ form: 'login' })(LoginForm);
