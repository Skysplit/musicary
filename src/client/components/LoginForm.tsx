import React, { PureComponent } from 'react';
import { FormikProps, FormikValues } from 'formik';
import { TextField, Button, Checkbox } from 'react-md';

interface LoginFormValues extends FormikValues {
  email: string;
  password: string;
}

type LoginFormProps = FormikProps<LoginFormValues>;

export default class LoginForm extends PureComponent<LoginFormProps> {
  handleTextFieldChange = (value: string, event: any) => (
    this.props.handleChange(event)
  )

  handleCheckboxChange = (checked: boolean, event: any) => (
    this.props.handleChange(event)
  )

  render() {
    const { handleSubmit, errors, values } = this.props;
    return (
      <form onSubmit={handleSubmit}>
        <TextField
          onChange={this.handleTextFieldChange}
          id="email"
          error={!!errors.email}
          errorText={errors.email}
          label="E-mail"
          value={values.email}
          fullWidth
        />

        <TextField
          onChange={this.handleTextFieldChange}
          id="password"
          label="Password"
          type="password"
          error={!!errors.password}
          errorText={errors.password}
          value={values.password}
          fullWidth
        />

        <Checkbox
          id="remember"
          name="remember"
          label="Remember me"
          onChange={this.handleCheckboxChange}
        />

        <Button type="submit" raised primary>
          Submit
        </Button>
      </form>
    );
  }
}
