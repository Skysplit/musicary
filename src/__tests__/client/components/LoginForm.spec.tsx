import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '@client/components/LoginForm';

describe('<LoginForm />', () => {
  const defaultProps = {
    handleChange: () => {},
    values: {
      email: 'test@email.com',
      password: 'testPassword',
    },
    errors: {
      email: 'error',
      password: 'error',
    },
  };
  test('should render component', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(<LoginForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
