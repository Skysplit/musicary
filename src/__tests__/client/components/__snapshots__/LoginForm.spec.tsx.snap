// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render component 1`] = `
<form>
  <Grid
    component="div"
    noSpacing={false}
    stacked={false}
  >
    <Cell
      component="div"
      desktopHidden={false}
      phoneHidden={false}
      size={12}
      tabletHidden={false}
    >
      <TextField
        error={true}
        errorText="error"
        fullWidth={true}
        id="email"
        label="E-mail"
        leftIconStateful={true}
        lineDirection="left"
        onChange={[Function]}
        passwordIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            remove_red_eye
          </FontIcon>
        }
        rightIconStateful={true}
        type="text"
        value="test@email.com"
      />
    </Cell>
  </Grid>
  <Grid
    component="div"
    noSpacing={false}
    stacked={false}
  >
    <Cell
      component="div"
      desktopHidden={false}
      phoneHidden={false}
      size={12}
      tabletHidden={false}
    >
      <TextField
        error={true}
        errorText="error"
        fullWidth={true}
        id="password"
        label="Password"
        leftIconStateful={true}
        lineDirection="left"
        onChange={[Function]}
        passwordIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            remove_red_eye
          </FontIcon>
        }
        rightIconStateful={true}
        type="password"
        value="testPassword"
      />
    </Cell>
  </Grid>
  <Grid
    component="div"
    noSpacing={false}
    stacked={false}
  >
    <Cell
      component="div"
      desktopHidden={false}
      desktopSize={6}
      phoneHidden={false}
      phoneSize={2}
      tabletHidden={false}
      tabletSize={4}
    >
      <Checkbox
        checkedIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            check_box
          </FontIcon>
        }
        id="remember"
        label="Remember me"
        name="remember"
        onChange={[Function]}
        uncheckedIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            check_box_outline_blank
          </FontIcon>
        }
      />
    </Cell>
    <Cell
      component="div"
      desktopHidden={false}
      desktopSize={6}
      phoneHidden={false}
      phoneSize={2}
      tabletHidden={false}
      tabletSize={4}
    >
      <withInk(withTooltip(Button))
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        primary={true}
        raised={true}
        type="submit"
      >
        Submit
      </withInk(withTooltip(Button))>
    </Cell>
  </Grid>
</form>
`;
